engine/src/main/java/org/terasology/rendering/dag/RenderGraph.java

```diff
@@ -91,19 +91,18 @@ public boolean connect(Node ... nodeList) {
         for (Node toNode : nodeList) {
             Preconditions.checkNotNull(toNode, "toNode cannot be null!");
 
-            if (fromNode == null) {
-                fromNode = toNode;
-                continue;
-            }
+            if (fromNode != null) {
+                boolean success = edgeMap.put(fromNode, toNode);
+                if (success) {
+                    reverseEdgeMap.put(toNode, fromNode);
+                } else {
+                    logger.warn("Trying to connect two already connected nodes, " + fromNode.getClass() + " and " + toNode.getClass());
+                }
 
-            boolean success = edgeMap.put(fromNode, toNode);
-            if (success) {
-                reverseEdgeMap.put(toNode, fromNode);
-            } else {
-                logger.warn("Trying to connect two already connected nodes, " + fromNode.getClass() + " and " + toNode.getClass());
+                returnValue = returnValue && success;
             }
 
-            returnValue = returnValue && success;
+            fromNode = toNode;
```

Comment:
Note that the logic was that every pair of adjecent nodes in the list becomes a fromNode, toNode and a connection is established between them, and after iterating through all pairs, the entire list is completed. However, I forgot the mark current toNode as the next fromNode, which lead to some.. "interesting" results.

I've added that line, and refactored the method a little.